scene/gui/slider.cpp

```diff
@@ -189,6 +189,35 @@ void Slider::_notification(int p_what) {
 			Size2i size = get_size();
 			double ratio = Math::is_nan(get_as_ratio()) ? 0 : get_as_ratio();
 
+			if (find_ticks) {
+				float step = get_step(), dist = get_max() - get_min();
+				int idist = dist;
+				if (step > dist) {
+					ticks = 0;
+					WARN_PRINT_ONCE("The step must be less than or equal to : " + itos(dist));
+				} else if (step == dist || step < 1) {
+					ticks = 0;
+				} else if (step >= 1) {
+					if (!Math::is_equal_approx(Math::fmod(dist, step), 0)) {
+						List<int> factors = List<int>();
+						for (int i = Math::sqrt(dist); i > 0; i--) { // find all divisors of the distance
+							if (idist % i == 0) {
+								factors.push_back(i);
+								factors.push_back(idist / i);
+							}
+						}
+						factors.sort();
+						for (int i = factors.size() - 1; i >= 0; i--) {
+							if (step >= factors[i]) {
+								set_step(factors[i]);
```

Comment:
It shouldn't change the number of steps. This becomes too convoluted if we have properties indirectly affect each other like that.

I also don't see this being useful in principle, because you want your users to be able to adjust the values with the precision specified in steps, physical limitations should be handled differently.